### PR TITLE
Doc improvement: Adding some content for Apache Kafka components

### PR DESCRIPTION
**Pull request by matzew**
## Proposed Changes

  * Adding some better docs for KafkaSource/Channel, since the READMEs were deeply hidden

## Follow up changes:

* move sample for the source _out_ to the `knative/docs` repo.  
* add Kafka channel samples to the `knative/docs` repo

/cc @samodell @joaedwar

### CODE REVIEW

File: kafka/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Apache Kafka Componentes for Knative
+
+Apache Kafka Componentes for Knative support the `KafkaChannel` and the `KafkaSource` implementation, which offers different ways to integrate Knative with the [Apache Kafka](http://kafka.apache.org/) broker.
```
Comment (joaedwar) [inline kafka/README.md]:
```suggestion
Apache Kafka Components for Knative support the `KafkaChannel` and the `KafkaSource` implementation, which offer ways to integrate Knative with [Apache Kafka](http://kafka.apache.org/) broker.
```

File: kafka/channel/README.md [hunk 2]
```diff
@@ -5,21 +5,21 @@ topics.
 
 ## Deployment steps
 
-1. Setup [Knative Eventing](../../../DEVELOPMENT.md)
+1. Setup [Knative Eventing](../../DEVELOPMENT.md)
 1. If not done already, install an Apache Kafka cluster!
```
Comment (joaedwar) [inline kafka/channel/README.md]:
```suggestion
Install an Apache Kafka cluster, if you have not done so already.
```

**Comment by matzew** [comment]
Not sure what that actually is:

```
=============================
==== RUNNING BUILD TESTS ====
=============================
==== Tue Aug 20 02:02:31 PDT 2019
=============================
----------------------------------------------
---- Checking links in the markdown files ----
----------------------------------------------
---- Tue Aug 20 02:02:31 PDT 2019
----------------------------------------------
README.md
	ERROR	https://goreportcard.com/badge/knative/eventing-contrib
		Dialing to the given TCP address timed out
	ERROR	https://goreportcard.com/report/knative/eventing-contrib
		Dialing to the given TCP address timed out
--------------------------------------
---- Checking that go code builds ----
--------------------------------------
---- Tue Aug 20 02:02:36 PDT 2019

```

File: kafka/README.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# Apache Kafka Componentes for Knative
+
+Apache Kafka Components for Knative support the `KafkaChannel` and the `KafkaSource` implementation, which offer ways to integrate Knative with the [Apache Kafka](http://kafka.apache.org/) broker.
+topics.
+
```
Comment (aslom) [inline kafka/README.md]: typo remove hanging "topcis."?

**Pull request merged**